frontend/packages/console-shared/src/components/editor/YAMLEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import Measure from 'react-measure';
+import MonacoEditor from 'react-monaco-editor';
+
+import { registerYAMLinMonaco, defaultEditorOptions, hackyFocusEditor } from './yaml-editor-utils';
+import ShortcutsLink from './ShortcutsLink';
+import SidebarLink from './SidebarLink';
+import './YAMLEditor.scss';
+
+interface YAMLEditorProps {
+  value?: string;
+  options?: object;
+  showSidebar?: boolean;
+  showShortcuts?: boolean;
+  minHeight?: string;
+  onChange?: (newValue, event) => {};
+  onSave?: () => {};
+  onResize?: () => {};
+  onToggleSidebar?: () => {};
+}
+
+const YAMLEditor = React.forwardRef<MonacoEditor, YAMLEditorProps>((props, ref) => {
+  const {
+    value,
+    options = defaultEditorOptions,
+    showShortcuts,
+    showSidebar,
+    onChange,
+    onSave,
+    onResize,
+    onToggleSidebar,
+    minHeight,
+  } = props;
+
+  const editorDidMount = (editor, monaco) => {
+    editor.layout();
+    editor.focus();
+    registerYAMLinMonaco(monaco);
+    monaco.editor.getModels()[0].updateOptions({ tabSize: 2 });
+    onSave && editor.addCommand(monaco.KeyMod.CtrlCmd | monaco.KeyCode.KEY_S, onSave); // eslint-disable-line no-bitwise
+  };
+
+  return (
+    <>
+      <div className="ocs-yaml-editor__links">
```

Comment:
Only render this if there are actions to show.

Reply:
Done.